src/test/java/it/gov/pagopa/payment/controller/BasePaymentControllerIntegrationTest.java

```diff
@@ -536,6 +538,26 @@ protected void configureMocks() {
             extractResponse(confirmPayment(trxCreated, MERCHANTID, ACQUIRERID), HttpStatus.BAD_REQUEST, null);
         });
 
+        //useCase 18: user cancel payment instead of authorizing
+        useCases.add(i -> {
+            TransactionCreationRequest trxRequest = TransactionCreationRequestFaker.mockInstance(i);
+            trxRequest.setInitiativeId(INITIATIVEID);
+
+            // Creating transaction
+            TransactionResponse trxCreated = createTrxSuccess(trxRequest);
+
+            // Relating to user
+            AuthPaymentDTO preAuthResult = extractResponse(preAuthTrx(trxCreated, USERID, MERCHANTID), HttpStatus.OK, AuthPaymentDTO.class);
+            assertEquals(SyncTrxStatus.IDENTIFIED, preAuthResult.getStatus());
+            checkTransactionStored(preAuthResult, USERID);
+
+            extractResponse(unrelateTrx(trxCreated, USERID), HttpStatus.OK, null);
+
+            TransactionInProgress stored = checkIfStored(trxCreated.getId());
```

Comment:
the unrelate operation should rollback the preAuth op:
you could check it retrieving the stored TransactionInProgress before the preAuth and assertEquals to the stored trx after unrelate (except timestamp fields)